include/linker/common-rom.ld

```diff
@@ -100,8 +100,17 @@
 		_bt_services_end = .;
 	} GROUP_LINK_IN(ROMABLE_REGION)
 
+#if defined(CONFIG_SETTINGS)
+	SECTION_DATA_PROLOGUE(_settings_handler,,SUBALIGN(4))
+	{
+		_settings_handler_list_start = .;
+		KEEP(*(SORT_BY_NAME("._settings_handler.static.*")))
+		_settings_handler_list_end = .;
+	} GROUP_LINK_IN(ROMABLE_REGION)
+#endif
+
 #if defined(CONFIG_BT_SETTINGS)
-	SECTION_DATA_PROLOGUE(_bt_settings_area,,SUBALIGN(4))
+	SECTION_DATA_PROLOGUE(_settings_area,,SUBALIGN(4))
```

Comment:
This seems wrong, the bt_settings_area has nothing to do with the settings handlers, it has a different type bt_settings_handler.

If the idea is to have just one handler and reuse on bluetooth side it should not be conditional to CONFIG_BT_SETTING, instead it should probably be CONFIG_SETTINGS and then remove the bt_settings_handler altogether.

Reply:
Yes, this is wrong, it entered during development and I have forgotten to correct it back to `_bt_settings_area`. Thanks for catching this.

In the long run, I would expect the handler registration for bt_settings to disappear, I would keep the rework to one settings_handler region for a future PR.